src/utils/helpers/queryHelper.js

```diff
@@ -0,0 +1,415 @@
+    });
+    return results;
+  }
+  if (userRole === 'Requester') {
+    const query =
+      'SELECT trips.departure,trips.destination,trips."travelReason",trips."dateOfTravel",trips."dateOfReturn"' +
+      ',users.id as "userId",users.firstname,users.lastname,users.email' +
+      ',accomodations.name as "accommodation" ' +
+      'FROM trips ' +
+      'LEFT JOIN users ON trips."userId"=users.id ' +
+      'LEFT JOIN accomodations ON trips."accomodationId"=accomodations.id ' +
+      `WHERE LOWER(array_to_string(destination, ',')) LIKE:destination` +
+      ` AND trips."userId"='${user}' `;
+    const [results] = await sequelize.query(`${query}`, {
+      replacements: { destination: `%${search}%` }
+    });
+    return results;
+  }
+};
+const ByDuration = async (duration, role, user) => {
+  if (role === 'Requester') {
+    const query =
+      'SELECT trips.departure,trips.destination,trips."travelReason",trips."dateOfTravel",trips."dateOfReturn"' +
+      ',users.id as "userId",users.firstname,users.lastname,users.email' +
+      ',accomodations.name as "accommodation" ' +
+      'FROM trips ' +
+      'LEFT JOIN users ON trips."userId"=users.id ' +
+      'LEFT JOIN accomodations ON trips."accomodationId"=accomodations.id ' +
+      `WHERE ("dateOfReturn"::DATE - "dateOfTravel"::DATE) = '${duration}' AND trips."userId"='${user}'`;
+    const [results] = await sequelize.query(`${query}`);
+    return results;
+  }
+  if (role === 'Travel Administrator') {
+    const query =
+      'SELECT trips.departure,trips.destination,trips."travelReason",trips."dateOfTravel",trips."dateOfReturn"' +
+      ',users.id as "userId",users.firstname,users.lastname,users.email' +
+      ',accomodations.name as "accommodation" ' +
+      'FROM trips ' +
+      'LEFT JOIN users ON trips."userId"=users.id ' +
+      'LEFT JOIN accomodations ON trips."accomodationId"=accomodations.id ' +
+      `WHERE ("dateOfReturn"::DATE - "dateOfTravel"::DATE) = '${duration}'`;
+    const [results] = await sequelize.query(`${query}`);
+    return results;
+  }
+  if (role === 'Manager') {
+    const query =
+      'SELECT trips.departure,trips.destination,trips."travelReason",trips."dateOfTravel",trips."dateOfReturn"' +
+      ',users.id as "userId",users.firstname,users.lastname,users.email' +
+      ',accomodations.name as "accommodation" ' +
+      'FROM trips ' +
+      'LEFT JOIN users ON trips."userId"=users.id ' +
+      'LEFT JOIN accomodations ON trips."accomodationId"=accomodations.id ' +
+      `WHERE ("dateOfReturn"::DATE - "dateOfTravel"::DATE) = '${duration}'` +
+      ` AND trips."accomodationId" IN ` +
+      `(SELECT id FROM accomodations WHERE "managerId"='${user}')`;
+    const [results] = await sequelize.query(`${query}`);
+    return results;
+  }
+};
+export {
+  tripSearchByName,
+  searchByEmail,
+  byDeparture,
+  byStartDate,
+  byStatus,
+  byDepartureAndDestination,
+  byDestination,
+  ByDuration
```

Comment:
What if the user is not among those userRole you specified like super admin you should put the else case for this when the user is like admin or anything else